### PR TITLE
[SSO] Fall back to Display Name if First and Last names are blank

### PR DESCRIPTION
**Pull request by biyeun**
## Summary
From QA we learned that it was possible for a new user to join and have their First and Last names be blank. This is because Azure AD does not make these fields required by default. However, Display Name is set to required.

This is a simple fix to ensure that if a First/Last name isn't available, we fall back to the Display Name

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Extremely well-tested area and new tests were added for this change.

### QA Plan
Addresses a QA bug.

### Safety story
Area is well tested and this is code that is presently not active on production, so it can be merged immediately after approval.

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/sso/tests/test_backends.py [hunk 1]
```diff
@@ -254,6 +254,81 @@ def test_new_user_created_and_data_is_saved(self):
             ]
         )
 
+    def test_new_user_displayname_is_used_if_first_and_last_are_missing(self):
```
Comment (mjriley) [inline corehq/apps/sso/tests/test_backends.py:257]: These new tests have a lot of code just to test that the names are grabbed appropriately. One way to shrink the tests and expose just the relevant information would be to create helper methods for the irrelevant details (i.e. the contents of `cleaned_data` don't matter for these tests). 

However, I think a better approach might be to make these tests lower level. It's great that there is a happy-path test to ensure that all the components integrate with each other, but I don't think its necessary to need an `AsyncSignupRequest` when we're testing edge cases within `get_sso_user_first_name_from_session`. 

With integration tests, beyond being more expensive, a developer needs full context to understand what is going on. The location of where those tests are, and what they should cover is ambiguous. If these tests were instead replaced with unit tests, then those problems are solved:
- I can look exclusively at `get_sso_user_first_name_from_session()` to understand the test.
- because `get_sso_user_first_name_from_session()` lives in _session_helpers.py_, it is natural that its tests live in _test_session_helpers.py_
- There is no ambiguity on what should be tested. If there is logic within `get_sso_user_first_name_from_session()`, a test is expected to verify that logic.

File: corehq/apps/sso/utils/session_helpers.py [hunk 2]
```diff
@@ -31,16 +31,33 @@ def _get_saml_user_data_property(request, prop_slug):
     return value
 
 
+def _get_display_name_from_session(request):
```
Comment (mjriley) [inline corehq/apps/sso/utils/session_helpers.py:34]: Is it possible to remove the dependence on request from this and other methods. In particular, all of these functions are named: `_from_session` rather than `_from_request`, implying that the request isn't necessary. Beyond that, we don't really care that its a formal 'session' object either -- it seems all that is necessary is that it has a dictionary-like interface, so we can call `.get()` on it. It would be nice if we could reduce these down to functions where I could just pass in a dict with the properties I cared about when testing them.

**Review by mjriley — APPROVED** [review]
Definitely agree that the PR will solve the issue, and I love that you tackled both edge cases of having only one name or a multi-word last name. I do think, with the complexity of these tests, however, they might be hard to maintain for someone else

**Comment by biyeun** [comment]
@mjriley taking note of the tests for a followup PR
